driver-core/src/test/functional/com/mongodb/internal/operation/ChangeStreamOperationSpecification.groovy

```diff
@@ -487,6 +487,7 @@ class ChangeStreamOperationSpecification extends OperationFunctionalSpecificatio
         when:
         expected = insertDocuments(helper, [5, 6])
         helper.killCursor(helper.getNamespace(), cursor.getWrapped().getServerCursor())
+        sleep(5000)
```

Comment:
Do you have a sense as to why this sleep helps?  What are we waiting to happen?  Does the killCursors not actually happen synchronously on the server?

Reply:
I think the issue is with the next and clean cycle not catching all the data.

I think I can probably revert the tryMultipleTimes change and only use that in change stream tests and then here  do something similar eg collect the results from nextAndClean until it reaches the expected size or it fails after multiple times.